### ISSUE TITLE
consensus reactor: making votes from commit

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#3596]
In #3298 we seek to explicitly exercise code paths necessary to reconstruct Vote from CommitSig even before we reduce the fields in CommitSig (ala #1648). 

If CommitSig does not include the validator address directly, this requires loading the current validator set, which leads to a question of how to handle an error if there is one, which there shouldn't be - see https://github.com/tendermint/tendermint/pull/3298/files#diff-f50d554d9c7f92cf7686b247b8c91419R642.

Alternatively, we may need to keep the addresses in the CommitSig to support light clients properly, which should eliminate the need to load validator sets, but it means 2k less in savings in block space for 100 validators (ie. 20byte addresses).

**Comment by ebuchman** [comment]
> Alternatively, we may need to keep the addresses in the CommitSig to support light clients properly, which should eliminate the need to load validator sets, but it means 2k less in savings in block space for 100 validators (ie. 20byte addresses).

Actually I'm not sure this has to go in the block itself if full nodes provide the lite clients with addresses for each commit. But maybe it's really worth the convenience of having it there.

**Comment by melekes** [comment]
> Alternatively, we may need to keep the addresses in the CommitSig to support light clients properly, which should eliminate the need to load validator sets, but it means 2k less in savings in block space for 100 validators (ie. 20byte addresses).

**TLDR** I don't think it's worth keeping the addresses since light clients will need additional info anyway.

Both current and Jae's implementation of the light client load `Validators` and `NextValidators` separately using `/validators` rpc method. Even if we decide to include addresses, light client will still need `NextValidators` and additional info that comes with `Validators` (VotingPower) to determine such things as `totalVotingPower`.

**Comment by ebuchman** [comment]
So light client always needs NextValidators, but in the bisecting case, it doesn't need the current Validators for the header it's trying to jump to. If we remove the addresses from the vote, then it will need to download those current validators, which seems kind of wasteful (all it really needs to know is if +1/3 of the validator set it already knows about signed this new commit, regardless of what the commit's actual validator set is).

So I think this is a case where leaving it in has some clear efficiency benefits to the lite client. And there's also the ergonomics and convenience of having them in there. I feel like by the time we'd want to remove them for full node efficiency, we'd be ready to signature aggregation in votes, which seems like a good time to deal with these things together (then lite clients will need to download validator sets again, but the commits will be much smaller).

**Comment by marbar3778** [comment]
I don't believe there are actionable steps within this issue? seems there was an agreement to keep the address within commitsig

**Comment by ebuchman** [comment]
Yeh I was wrong in my last comment, but I think we're going to keep it in anyways, as per the latest state of the ADR. So I think we can close this.
